tap_logic4/streams.py

```diff
@@ -191,20 +190,53 @@ class ProductsStream(Logic4Stream):
         th.Property("VendorCode", th.StringType),
         th.Property("ProductTemplateId", th.IntegerType),
         th.Property("ProductTemplateName", th.StringType),
+        th.Property("IsVisibleOnWebShop", th.BooleanType),
+        th.Property("IsVisibleInLogic4", th.BooleanType),
     ).to_dict()
-
+    
+    is_visible_pairs = [
+        {"IsVisibleOnWebShop": True, "IsVisibleInLogic4": True},
+        {"IsVisibleOnWebShop": False, "IsVisibleInLogic4": True},
+        {"IsVisibleOnWebShop": True, "IsVisibleInLogic4": False},
+        {"IsVisibleOnWebShop": False, "IsVisibleInLogic4": False},
+    ]
+    is_visible_current_pair = None
+    
     def prepare_request_payload(self, context, next_page_token):
-        payload = super().prepare_request_payload(context, next_page_token)
-        if "IsVisibleOnWebShop" in self.config:
-            payload["IsVisibleOnWebShop"] = self.config["IsVisibleOnWebShop"]
-
-        if "IsVisibleInLogic4" in self.config:
-            payload["IsVisibleInLogic4"] = self.config["IsVisibleInLogic4"]
+        payload = super().prepare_request_payload(context, None)
+        if next_page_token and next_page_token.get("counter"):
+            payload["SkipRecords"] = next_page_token.get("counter")
+        if self.is_visible_pairs:
+            self.is_visible_current_pair = self.is_visible_pairs.pop()
```

Comment:
I think we should only go to the next value in `is_visible_pairs` if we're done paginating one of the values and counter is 0, rn it looks like we go to the next value in each iteration even if we're not done paginating one of the values.

Reply:
iteration is working good 👍 , just one thing the first request will go without any filter for `is_visible_current_pair` as the `next_page_token` is None in the first iteration